File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 
function twoSum(nums, target) {
    const numIndicesMap = new Map();

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];

        if (numIndicesMap.has(complement)) {
            return [numIndicesMap.get(complement), i];
        }

        numIndicesMap.set(nums[i], i);
    }

    // If no solution is found
    return [];
}

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        //Create a HashMap to store he number and its index
        HashMap<Integer, Integer> map = new HashMap<>();
        
        //Traverse the array
        for(int i = 0; i<nums.length; i++){
            int complement = target - nums[i];
            if(map.containsKey(complement)){
                return new int[] {map.get(complement), i};
            }
            map.put(nums[i], i);
        }
        return new int[0];
        
    }
}